kienstra/headless-blocks
# Add filters to allow front-end rendering of blocks

This is ready for review.

## Changes
* In the block editor, allows an alternate preview component for GCB blocks:
<img width="717" alt="block-with-alternate-preview" src="https://user-images.githubusercontent.com/4063887/99588917-e3565180-29b0-11eb-9a28-cff4a469ec5c.png">

* On the front-end, renders the block as only an HTML comment, so it can be rendered with the same React component shown above, accepting `attributes` as `props`

<img width="998" alt="the-html-comments" src="https://user-images.githubusercontent.com/4063887/99575170-e136c780-299d-11eb-83a7-fed7d76d80ca.png">

## Testing Instructions
1. `checkout` this WordPress plugin PR: https://github.com/studiopress/genesis-custom-blocks/pull/46
1. `$ cd genesis-custom-blocks && npm install && composer install && npm run dev`
1. `clone` this PR's repo into `wp-content/plugins/` and `checkout` this PR's branch
1. `$ cd headless-blocks # cd into this repo`
1. Create a new GCB block 
1. Name it 'example': <img width="985" alt="create-new-block" src="https://user-images.githubusercontent.com/4063887/99585234-ad629e80-29ab-11eb-9317-ac5b90039cf6.png">
1. Add a field
1. Don't change anything in the field, keep the name of `'new-field'`: <img width="1036" alt="abiu-here" src="https://user-images.githubusercontent.com/4063887/99585516-0e8a7200-29ac-11eb-9b7b-919db584bc76.png">
1. Publish the block
1. Create a new post
1. Add that 'Example' block you created: <img width="847" alt="add-the-example-block" src="https://user-images.githubusercontent.com/4063887/99585658-3c6fb680-29ac-11eb-9356-331d138e8c06.png">
1. Type something in the `<input>`: <img width="858" alt="type-something" src="https://user-images.githubusercontent.com/4063887/99585750-5f9a6600-29ac-11eb-8137-79a59177ea77.png">
1. Click outside of the block to see the preview
1. Expected: the alternate component displays: <img width="889" alt="vie-bsdf" src="https://user-images.githubusercontent.com/4063887/99586058-cd469200-29ac-11eb-9263-7fe478461e89.png">
1. Go to the front-end preview of the post
1. Expected: There's no visible markup in the post content <img width="1109" alt="no-markup-in" src="https://user-images.githubusercontent.com/4063887/99587068-2ebb3080-29ae-11eb-880f-f1590c8466ba.png">
1. Expected: The block appears in an HTML comment, with the serialized attributes: 

```html

```

Update: There's now a component to render that comment as a React component
